### PR TITLE
Openbsd fixes

### PR DESCRIPTION
**Pull request by vext01**
Fixes various bits on OpenBSD.

You probably want to look at individual commits.

OK?

### CODE REVIEW

File: krun/platform.py [hunk 1]
```diff
@@ -696,8 +697,13 @@ def pin_process_args(self):
         return []  # not supported on OpenBSD
 
     def is_virtual(self):
-        """Not yet implemented on this platform, assume not virtualised"""
+        """If we see a vio(4) disk in the dmesg, this is a virtual machine"""
 
+        dmesg_lines = self._collect_dmesg_lines()
+        comp_pat = re.compile(OpenBSDPlatform.VIO_DMESG_PATTERN)
+        for line in dmesg_lines:
```
Comment (ltratt) [inline krun/platform.py]: Might be simpler as something like `return any(comp_pat.match(l) for l in dmesg_lines)`?

**Comment by ltratt** [comment]
One minor comment but otherwise looks OK to me.

**Comment by ltratt** [comment]
We have a build failure, although it's not immediately obvious to me what's failed.

File: krun/platform.py (hunk 1 above)
Comment (vext01) [reply krun/platform.py]: Simpler to read, much less efficient. Thoughts?

**Comment by vext01** [comment]
a sleep(1) waited for 0.999626 seconds. I had asserted that we always wait at-least a second.

I didn't think this was possible!

File: krun/platform.py (hunk 1 above)
Comment (ltratt) [reply krun/platform.py]: Is this a performance critical part of things? [I think not, but I might be wrong!]

Comment (vext01) [reply krun/platform.py]: Well, not hugely, it's just a waste of cycles, and regex aint exactly efficient.

My dmesg is 528 lines long. That's a lot of regex.

Comment (ltratt) [reply krun/platform.py]: I'd rather have easy to read code: you'd be amazed at how quick that code will execute on a modern machine!

Comment (vext01) [reply krun/platform.py]: I'm well aware of how fast a machine is :P 

I don't feel strongly enough to argue more :) Fixed.

**Comment by vext01** [comment]
Can I squash this?

File: krun/platform.py (hunk 1 above)
Comment (ltratt) [reply krun/platform.py]: Oh, I see what you were implying. You needn't worry :)

```python
>>> type(x for x in [1])
<type 'generator'>
>>> type([x for x in [1]])
<type 'list'>
```

**Comment by ltratt** [comment]
I've always assumed that `sleep` is an approximation that one should not rely on (e.g. does it use a monotonic timer?). I'm not convinced that asserting anything about its duration is sensible to be honest...

File: krun/platform.py (hunk 1 above)
Comment (vext01) [reply krun/platform.py]: Um. Not sure what you mean. I just didn't want to do load of regex and allocate a big list if it aint necessary. The vio entry, if it exists, will be early in the dmesg.

If it makes me look cleverer, i'll take credit for your code snippet though :)

**Comment by vext01** [comment]
It is the monotonic timer, fwiw.

File: krun/platform.py (hunk 1 above)
Comment (ltratt) [reply krun/platform.py]: What I mean is that the version I'm supplying lazily evaluates the list, so its performance is the same as the `for` loop. So, I believe, it's smaller code that's (more or less) the same performance.

You're welcome to take the credit though -- you'll also get the blame ;)

Comment (vext01) [reply krun/platform.py]: Ah, the generator doesn't allocate the list?

**Comment by ltratt** [comment]
[I also don't see any reference to a monotonic timer in `sleep` or `nanosleep` on OpenBSD.]

**Comment by vext01** [comment]
OK, I see where the confusion is.

This test is like this:
```
t1 = monotonic_time()  # monotonic
sleep(1)  # who knows how it works...
t2 = montonic_time()  # monotonic

interval = t2 - t1
```

I was talking about the readings, you are talking about the sleep.

So about exiting early:
```
     The sleep() function suspends execution of the calling process until
     either seconds seconds have elapsed or a signal is delivered to the
     process and its action is to invoke a signal-catching function or to
     terminate the process.  The suspension time may be longer than requested
     due to the scheduling of other activity by the system.
```

So as i see it, unless the process is killed then the wait should be longer than the requested interval.

We know the process wasn't killed, as it exited with 0.

This is weird.

**Comment by vext01** [comment]
Oh, the process doesn't have to be dying... OK. Well, I don't know who'd be sending us a signal...

Shall we merge this then? I don't want to dwell on this too much.

**Comment by ltratt** [comment]
There are other signals than kill that might be received. Whatever, I don't think we can rely on `sleep` sleeping for a given period: a test that relies on it is going to be fragile.

**Comment by ltratt** [comment]
Agreed. I suggest we remove this particular part of the test.

**Comment by vext01** [comment]
Yeah, the comment in the test pretty much says this.

Let me make a better test.

**Comment by vext01** [comment]
This is more robust IMHO.

**Comment by ltratt** [comment]
Ok, I'm fine with that. Squash and then I'll merge?

**Comment by vext01** [comment]
splat.

**Comment by ltratt** [comment]
Ah, you also need to rebase against origin/master please.

**Comment by vext01** [comment]
Oops. And, like so. Go ahead.

**Pull request merged**